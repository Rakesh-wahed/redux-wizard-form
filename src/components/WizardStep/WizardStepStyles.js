import styled from 'styled-components';

export const Wrapper = styled.form`
  display: inline-block;
  width: 100%;
`;
