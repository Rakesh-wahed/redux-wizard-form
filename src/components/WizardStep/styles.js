import styled from 'styled-components';

export const Wrapper = styled.form`
  display: flex;
  justify-content: center;
  width: 100%;
`;
