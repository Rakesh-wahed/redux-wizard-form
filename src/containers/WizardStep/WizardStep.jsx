import { connect } from 'react-redux';
import { reduxForm, getFormMeta, getFormSyncErrors } from 'redux-form';
import { WizardStep } from '../../components';

export const reduxWizardStep = formOptions => {
  const reduxFormInstance = reduxForm({
    ...formOptions,
    destroyOnUnmount: false,
    forceUnregisterOnUnmount: true
  });

  return reduxFormInstance(
    connect(
      state => ({
        meta: getFormMeta(formOptions.form)(state),
        errors: getFormSyncErrors(formOptions.form)(state)
      }),
      null
    )(WizardStep)
  );
};
