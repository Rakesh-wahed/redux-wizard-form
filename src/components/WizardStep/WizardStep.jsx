import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { reduxForm } from 'redux-form';
import { Wrapper } from './styles';

export class WizardStepComponent extends Component {
  constructor(props) {
    super(props);
    const { formOptions, component, onSubmit, name, addStepName, onWizardComplete, stepsSize, step } = props;
    const InnerComponent = component;

    this.WizardStepForm = reduxForm(formOptions)(formProps => (
      <Wrapper onSubmit={formProps.handleSubmit(step === stepsSize - 1 ? onWizardComplete : onSubmit)}>
        <InnerComponent {...formProps} />
      </Wrapper>
    ));

    addStepName(name);
  }
  render() {
    const { WizardStepForm } = this;
    const { currentStep, step } = this.props;

    if (currentStep >= step) {
      return  <WizardStepForm />;
    }

    return null;
  }
}

WizardStepComponent.propTypes = {
  name: PropTypes.string,
  currentStep: PropTypes.number.isRequired,
  step: PropTypes.number.isRequired,
  stepsSize: PropTypes.number.isRequired,
  formOptions: PropTypes.shape({
    form: PropTypes.string.isRequired,
    onChange: PropTypes.func,
    onSubmit: PropTypes.func,
    onSubmitFail: PropTypes.func,
    onSubmitSuccess: PropTypes.func
  }).isRequired,
  component: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onWizardComplete: PropTypes.func.isRequired,
  addStepName: PropTypes.func.isRequired
};

WizardStepComponent.defaultProps = {
  name: 'Step'
};
