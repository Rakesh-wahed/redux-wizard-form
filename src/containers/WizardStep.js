import { connect } from 'react-redux';
import { formSubmit } from '../actions';
import { getFormOptions } from '../reducer';
import { WizardStepComponent } from '../components/WizardStep/WizardStep.jsx';

export const WizardStep = connect(
  state => ({
    formOptions: getFormOptions(state)
  }),
  dispatch => ({
    onSubmit: data => dispatch(formSubmit(data))
  })
)(WizardStepComponent);
