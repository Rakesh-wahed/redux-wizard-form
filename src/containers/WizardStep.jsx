// @flow
import { connect } from 'react-redux';
import { nextStep, wizardStepNameAdd, wizardComplete } from '../actions';
import { getFormOptions, getCurrentStep, getStepsSize } from '../selectors';
import { WizardStepComponent } from '../components/WizardStep/WizardStep';

export const WizardStep = connect(
  (state: Object): Object => ({
    formOptions: getFormOptions(state),
    currentStep: getCurrentStep(state),
    stepsSize: getStepsSize(state)
  }),
  (dispatch: Function): Object => ({
    addStepName: (stepName: string): void =>
      dispatch(wizardStepNameAdd(stepName)),
    onSubmit: (): void => dispatch(nextStep()),
    onWizardComplete: (data: Object): void => dispatch(wizardComplete(data))
  })
)(WizardStepComponent);
