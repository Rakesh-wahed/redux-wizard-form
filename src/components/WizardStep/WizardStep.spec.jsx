import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { WizardStepComponent } from './WizardStep';

Enzyme.configure({ adapter: new Adapter() });

describe('WizardStep component', () => {
  test('snapshot', () => {
    const props = {
      currentStep: 0,
      step: 0,
      stepsSize: 1,
      formOptions: { form: 'formTest' },
      isWizardComplete: false,
      isLoaded: true,
      component: () => {},
      addStepName: () => {},
      onSubmit: () => {},
      onWizardComplete: () => {}
    };
    const Component = shallow(<WizardStepComponent {...props} />);
    expect(Component).toMatchSnapshot();
  });
});
