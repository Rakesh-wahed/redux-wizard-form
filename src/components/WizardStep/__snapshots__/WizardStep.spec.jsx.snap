// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WizardStep component snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WizardStepComponent
    addStepName={[Function]}
    component={[Function]}
    currentStep={0}
    formOptions={
      Object {
        "form": "formTest",
      }
    }
    isLoaded={true}
    isWizardComplete={false}
    name="Step"
    onSubmit={[Function]}
    onWizardComplete={[Function]}
    step={0}
    stepsSize={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
